src/System.Device.Gpio/System/Device/Spi/Devices/UnixSpiDevice.cs

```diff
@@ -71,7 +72,17 @@ private unsafe void Initialize()
                 int result = Interop.ioctl(_deviceFileDescriptor, (uint)SpiSettings.SPI_IOC_WR_MODE, nativePtr);
                 if (result == -1)
                 {
-                    throw new IOException($"Error {Marshal.GetLastWin32Error()}. Can not set SPI mode to {_settings.Mode}.");
+                    // We should try the other mode and invert the buffers manually
+                    // RPI can't open Mode0 in LSB for example
+                    mode = _settings.DataFlow == DataFlow.LsbFirst ? mode &= ~UnixSpiMode.SPI_LSB_FIRST : mode |= UnixSpiMode.SPI_LSB_FIRST;
```

Comment:
can you change this line to regular `if`? it's super confusing and you also double assign the value to `mode`